client.go

```diff
@@ -103,6 +90,7 @@ func (c *Client) NewSession(ctx context.Context, opts ...SessionOption) (*Sessio
 	var fr frames.Frame
 	select {
 	case <-ctx.Done():
+		// TODO: this will leak s
```

Comment:
Can we file an issue to fix this? I'm assuming this is related to the leak conversation we had the other day.